src/middlewared/middlewared/test/integration/utils/client.py

```diff
@@ -10,27 +10,39 @@
 from middlewared.client.utils import undefined
 
 __all__ = ["client", "host", "host_websocket_uri", "password", "session", "url", "websocket_url"]
+IS_HA = os.environ.get('SERVER_TYPE') == 'ENTERPRISE_HA'
 
 
 @contextlib.contextmanager
 def client(*, auth=undefined, auth_required=True, py_exceptions=True, log_py_exceptions=True, host_ip=None):
     if auth is undefined:
         auth = ("root", password())
 
+    uri = host_websocket_uri(host_ip)
     try:
-        with Client(host_websocket_uri(host_ip), py_exceptions=py_exceptions, log_py_exceptions=log_py_exceptions) as c:
+        with Client(uri, py_exceptions=py_exceptions, log_py_exceptions=log_py_exceptions) as c:
             if auth is not None:
                 logged_in = c.call("auth.login", *auth)
                 if auth_required:
                     assert logged_in
             yield c
-    except socket.timeout:
-        fail('socket timeout')
+    except socket.timeout as e:
+        if IS_HA:
+            # don't fail the entire run on HA systems since
+            # we perform failovers to and from each controller
+            # so timeouts could happen (which are expected)
```

Comment:
Are they expected on certain tests or random tests?

Reply:
That's hard to predict and is why I don't even like this functionality in the first place. I'd much rather just log to the active node on an HA machine then try and send duplicate messages across the heartbeat for no real gain. If anything, it's confusing because the standby node actually isn't doing _anything_ wrt to these tests.

Reply:
WHich tests? I don't understand. This fails the whole test suite if the connection to another node times out. There is no much sense to try to run each and every test against the dead node. Is HA VIP node expected to be down, but just for a while, and we assume it's fine?

Reply:
Oh I see, my branch is missing a commit.... one moment